README.rst

```diff
@@ -1,102 +1,208 @@
+  other generates. The language primitives are designed so that the order in
+  which the macros are expanded cannot affect their results, and indeed the
+  same is true for the order in which the macro expansion and type-inference
+  steps are interleaved. This makes Klister code more robust to refactoring.
+* `Problem-aware macros`_; in addition to the type, a macro can learn which
+  "problem" it needs to solve, namely whether it must generate an expression, a
+  type, a pattern, etc.
+* `Custom type-driven code generation`_, via macros which generate code from a
+  type.
+* Languages with `custom type systems`_, via macros which reinterpret types
+  into those of an existing ``#lang``, and which contribute to type inference
+  by providing type information about the code they generate. The variety of
+  type-systems which can be implemented this way is unforunately limited by
+  the core type system to which everything must be rewritten.
+* Languages with `custom implicit terms`_, via macros which generate terms of
+  an existing ``#lang`` based on a type in the new ``#lang``.
+
+.. _bibliography: bibliography.rst
+.. _Type-providing macros: TODO: write a small example demonstrating this
+   feature.
+.. _Type-aware macros: TODO: write a small example demonstrating this feature.
+.. _Stuck macros: TODO: write a small example demonstrating this feature. Maybe
+   the traverse-traverse-id example from Compose NYC 2019?
+.. _Problem-aware macros: TODO: write a small example demonstrating all the
+   different problems one can write a macro for.
+.. _Custom type-driven code generation: TODO write a small example
+   demonstrating the feature. Perhaps the traverse-traverse-id example again?
+.. _custom type systems: TODO: write an example #lang in which functions are
+   not curried, writing copious comments.
+.. _custom implicit terms: TODO: improve the comments in the
+   implicit-conversion example, then link to it.
+
+While we think Klister demonstrates some neat ideas, there are some limitations
+which make Klister an impractical choice for most real-life projects. If you
+want to help make Klister a more practical language, please `reach out`_!
+
+.. _reach out: https://github.com/gelisam/klister/issues/new
+
+Here are the most prominent Racket features which are missing from Klister:
+
+* Klister does not have reader macros, and thus every ``#lang`` looks like a
+  Lisp. This also limits languages to Integer literals and String literals.
+* `local-expand`_ is planned, but not yet implemented.
+* `Syntax parameters`_ are planned, but not yet implemented.
+
+.. _local-expand: https://github.com/gelisam/klister/issues/144#issuecomment-1133964551
+.. _Syntax parameters: https://github.com/gelisam/klister/issues/105
+
+Here are the most prominent Haskell features which are missing from Klister:
+
+* `Type classes` are planned, but not yet implemented.
+* Klister is strict, not lazy. Racket has a lazy ``#lang``, so it is probably
```

Comment:
This was an intentional design choice, not a missing feature :-)

Reply:
Suggestion: "The core language of Klister is very much like a strict Haskell 98, except it is missing the following features: "

Reply:
fixed in a much simpler way in 36163623ecc29197331db773cc2af3c37f11ccbd